### ISSUE TITLE
Allow pulling images from private repositories

### ISSUE THREAD

**Issue by erinmcgill** [kitematic/kitematic#232]
I am testing kitematic out for our DevOps group. I am trying to pull from our private repo one of the containers but it says "Cannot find a matching image". I know the image is correct because I copy/pasted from our running container in our QA environment.

*sad face*

**Comment by gscott** [comment]
+1 You'll need to add support for logging in to Docker Hub and seeing private repos there.

**Comment by robwithhair** [comment]
I am facing this same issue.  We're implementing a docker development workflow at our company and this tool would be a godsend, however I'm unable to see my private dockerhub repos so it won't work for us right now.

I have to say brilliant app.  I'm loving using it apart from the private repo issue.

**Comment by TravisAtBanjo** [comment]
Unfortunately, without the ability to log in and pull private repos, I'll have to hold off on really testing this out.

**Comment by mgmonteleone** [comment]
The ability to use private repos would be a very important addition. Perhaps you could point out where that would be done, and we could contribute that portion?

**Comment by funkytaco** [comment]
@mgmonteleone I think maybe here https://github.com/kitematic/kitematic/blob/master/src/Registry.js

**Comment by rsheldon** [comment]
How about supporting a list of repos that should be searched (dockerhub being the default). Then have something in the UI to differentiate which repo it came from?

**Comment by jeffdm** [comment]
Hey all - thanks for the issue and feedback.

This feature is heavily WIP by the team, but first needs the ability to login / signup.

Once that's done this should be relatively straightforward via `Registry.js`. Keep an eye out for work around login / signup.

**Comment by lorenwest** [comment]
This doesn't rely on login / signup for our internal repository, since it's locked down by LDAP credentials. All we need is the ability to enter the container name just like we do on command line docker:

```
docker pull artifactory.cobalt.com/hyde
```

Allowing us to enter it in the search bar would un-block our in house use of kitematic.  Searching & browsing the in-house repository would be nice, but they can wait on a future release (along with login)

**Comment by johnyzed** [comment]
In the meantime, you can do so with the command line provided by Kitematic ( docker icon in left down corner of the window). You run 'docker login', enter your credentials and after you can pull and run containers from your private repo. You can manage the running containers from the Kitematic UI.

**Comment by therealjessesanford** [comment]
bump

**Comment by balakumarpg** [comment]
this feature would help a lot +1.

**Comment by funkytaco** [comment]
>It's our highest priority feature!

I know you guys are busy but it doesn't seem like it is to me.

Anyone know where this is on the roadmap? I'm trying to get the team on board with Docker (they're Sys. Admins and Engineers, not devs) and this would really help.

**Comment by nritholtz** [comment]
+1

**Comment by abhsc** [comment]
+1 for being able add private *insecure* registries

**Comment by toebbel** [comment]
+1 for allowing private registries

**Comment by FrenchBen** [comment]
Looking at the above 'pool' of test users, can some of you QA this PR?
https://github.com/kitematic/kitematic/pull/1076

It should add support within the preference panel for any repo link. Only downfall is that if the repo isn't valid/online it will currently prevent Kitematic from loading properly. 

I don't know if there's a way to test the repo and confirm it's a proper docker repo? Any feedback/input welcome :D

**Comment by FrenchBen** [comment]
PR https://github.com/kitematic/kitematic/pull/1076 has been updated to add a private repo test, so that Kitematic has a valid repo to use/work from.

**Comment by davidkarlsen** [comment]
What's holding back on this? Is it because dockertoolbox should generate $$ for dockerhub - and dockerhub only?

**Comment by FrenchBen** [comment]
@davidkarlsen A lot of moving parts not at all related with $$  - Please see https://github.com/kitematic/kitematic/issues/210#issuecomment-156252142 for details.

Closing in favor of #210
